docs/gc.md

```diff
@@ -35,31 +35,34 @@ To prevent applications running out of memory, objects in the Java heap that are
 
 ## Garbage collection policies
 
-OpenJ9 provides several different GC policies, each one specialized for operation with a specific heap configuration and one or more collectors. These policies are summarized in the followng table. Follow the links to read more about individual policies:
+OpenJ9 provides several GC policies that are designed around different application workloads. Each policy operates on a specific heap configuration and one or more collectors. These policies are summarized in the following table. Follow the links to read more about individual policies:
```

Comment:
One comment that hopefully improves general understanding (not necessarily suggesting how introduction of this paragraphs should be edited, but should probably affect the content of the table later bellow, especially the last column)

I see a GC policy as a set of GC (and Allocator) characteristics and features to optimize one or more performance aspects of the application execution. 

Some important GC features:
- tracing or reference counting (almost all modern GCs are tracing, what indeed our GCs are, so this is typically not discussed)
- generational (with global and local/partial GCs) or non-generational (just global GC)
- if generational, it could be 2-generational or multi-generational
- style: mark-sweep(-compact) or evacuating, or in other simpler terms: non-moving or moving
- heap-organization, what is closely related to generational aspect: flat (non-generational), segregated for generations (Tenure and Nursery), multi-generational (with typically fixed sized regions)
- incremental or non-incremental, where a GC cycle (whether local or global) could be just one increment or series of discrete increments
- concurrent (GC increment or full cycles are done by background threads while application threads are still running) or non-concurrent (application is stopped, and GC increment or full cycle is exclusively run). This separately/idependability applies for local and global, if both exist.
- some more advanced characteristics are styles of concurrent GCs (write or read barrier based, and then there are various sub-styles of each).

Typically, a GC policy may encompass characteristics of Allocator as well, like:
- Free Memory Pool organization like: Address Ordered List or with Size Class Grouping
- Memory Pool search algorithm: First-Fit or Best-Fit or some hybrid

Reply:
Performance asspects are:
- application throughpup
- memory footprint
- average and/or worst case pause time
- startup time

Reply:
@amicic - I had an initial pass at this improvement to the overview, but haven't put it in the PR yet because I think it probably requires more work. (I didn't include all the detail you provided ... unsure if this is enough) Please provide some feedback:

## Garbage collection policies

OpenJ9 provides several GC policies that are designed around different application workloads. Each GC policy consists of a set of characteristics and features that aim to optimize one or more performance aspects of a running application. These performance aspects include application throughput, memory footprint, average pause times, worst-case pause times, and startup time. 

Different policies require a Java heap that is configured in different ways in order to achieve different goals. The following layouts exist: 

- A *flat* Java heap consists of one region of memory that contains a single generation of objects. With this type of heap, GC consists of a single global GC cycle that is triggered when the heap is full.
- A *generational* Java heap is segregated for different generations of objects, consisting of `tenure` and `nursery` regions. With this type of heap, objects can be moved between regions, depending on age and liveness. GC operations can then consist of local or partial GC cycles on objects that are no longer in use, thereby attempting to delay or avoid a global GC cycle.
- A *multi-generational* Java heap consists of multiple regions, typically of a fixed size, that support multiple generations of objects. Again, this type of arrangement can support local or partial GC cycles on objects that are no longer in use, attempting to delay or avoid a global GC cycles.

Different policies also employ different collection processes, which operate on areas of the heap in different ways. The following distinctions exist:

Global and local GC cycles can be incremental, where they occur in a series of discrete increments,  or non-incremental, where they occur in a single operation. These cycles can also be concurrent, where the operations take place while applications are still running, or non-concurrent, where the application is stopped (*stop-the-world* (STW)) while GC operations take place. More advanced characteristics exist for concurrent-style GC policies, including read or write barriers that are raised to allow application processing to continue without a STW collection taking place.

Reply:
This content is now in the PR